includes/Modules/Ads.php

```diff
@@ -60,6 +66,36 @@ final class Ads extends Module implements Module_With_Assets, Module_With_Debug_
 
 	const SCOPE = 'https://www.googleapis.com/auth/adwords';
 
+	/**
+	 * Input access abstraction.
+	 *
+	 * @since n.e.x.t
+	 * @var Input
+	 */
+	private $input;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Context        $context        Plugin context.
+	 * @param Options        $options        Optional. Option API instance. Default is a new instance.
+	 * @param User_Options   $user_options   Optional. User Option API instance. Default is a new instance.
+	 * @param Authentication $authentication Optional. Authentication instance. Default is a new instance.
+	 * @param Assets         $assets  Optional. Assets API instance. Default is a new instance.
+	 */
+	public function __construct(
+		Context $context,
+		Options $options = null,
+		User_Options $user_options = null,
+		Authentication $authentication = null,
+		Assets $assets = null
+	) {
+		parent::__construct( $context, $options, $user_options, $authentication, $assets );
+		$this->input = new Input();
+	}
```

Comment:
This isn't necessary (and also breaks DI) since Input is available via `Context->input()`

Reply:
Didn't see we have it there, thanks!